Coin config additions v2

Machotacoz opened the pull request.

Gumbees committed:
added zel.json

Gumbees committed:
added votecoin

Gumbees committed:
added vot.json pool_config

Gumbees committed:
zel didn't push to this branch under pool_configs.

Gumbees committed:
Corrected port and spacing

Spacing should be corrected. let me know if its still off. Looks good in notepad++.

Added correct RPC port for zel.json.

Gumbees committed:
spacing fix

Machotacoz committed:
snowgems coin perams

test on my pool and was able to accept blocks

Machotacoz committed:
snowgem pool config

Gumbees commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+	"daemon": {
+        "host": "127.0.0.1",
+        "port": 8232,
+        "user": "testuser",
```
Change testuser and testpass to rpcuser and rpcpassword. Trying to keep the config files on a standard.

Gumbees commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+	"maxBlocksPerPayment": 3,
+	"daemon": {
+        "host": "127.0.0.1",
+        "port": 8232,
```
Update rpc ports to their defaults. xsg i believe is 16112. they can be found on the coins github repo or usually their announcement page on bitcointalk. This will need done for the other pool_configs.

Gumbees reviewed (changes requested):
In code comments.

Gumbees commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+            "tls": false,
+            "varDiff": {
+                "minDiff": 0.04,
+                "maxDiff": 16,
```
change maxDiff to 36 for all pool_configs.

Gumbees commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+	"paymentMode": "prop",
+	"_comment_paymentMode":"prop, pplnt",
+	"paymentInterval": 20,
+	"minimumPayment": 0.1,
```
Change minimumPayment from 0.1 to 0.05.

Gumbees commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+        "port": 3306,
+        "user": "me",
+        "password": "mypass",
+        "database": "zec",
```
Change the database to the actual coin symbol for all pool_configs.

Gumbees reviewed (changes requested):
Can you also move all pool_config examples to pool_configs/examples? The last merge has this done. Thanks! Good work!

egyptianbman commented on coins/vot.json:
```diff
@@ -0,0 +1,9 @@
+    "requireShielding": true,
+    "peerMagic": "24e92764",
+    "txfee": 0.0004,
+    "overwinter": true
```
Can you move this up to below algorithm?

Gumbees replied:
@egyptianbman go ahead and make the changes. I want to add these asap.

egyptianbman commented on coins/xsg.json:
```diff
@@ -0,0 +1,38 @@
+    "percentFoundersReward": 5,
+    "maxFoundersRewardBlockHeight": 2106399,
+    "foundersRewardAddressChangeInterval": 105320.95,
+    "parameters": {
```
Let's put these below algorithm also. since this is related to the algorithm (same for other coins).

egyptianbman commented on coins/vot.json:
```diff
@@ -0,0 +1,9 @@
+    "symbol": "vot",
+    "algorithm": "equihash",
+    "requireShielding": true,
+    "peerMagic": "24e92764",
```
I actually don't have peerMagic for VOT. Does anyone know where it's derived from? I've tried looking for peermagic in the node codebase to no avail. I've always relied on searching to acquire it.

egyptianbman commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+{
```
This should go into the `pool_configs/examples` directory. You'll need to pull master in to get that.

egyptianbman commented on pool_configs/vot.json:
```diff
@@ -0,0 +1,87 @@
+
+    "tAddress": "",
+    "_comment_tAddress": "transparent address used to send payments, make this a different address, otherwise payments will not send",
+
```
Missing `invalidAddress`

egyptianbman reviewed (commented):
@Machotacoz @nettts would you guys mind if I made the updates from the comments I left and others myself? I don't want to step on anyone's toes.

zzzpotato committed:
peerMagic

zzzpotato committed:
Formatting. Moved parameters

zzzpotato committed:
Formatting. Moved parameters

zzzpotato committed:
Formatting. Added invalidAddress

zzzpotato committed:
Added invalidAddress. Small tweaks

zzzpotato committed:
Formatting. Added invalidAddress. Small tweaks

zzzpotato committed:
Fixed format. 

peerMagic seems to be the typical 24e92764 per the developers comment on this issue https://github.com/z-classic/z-nomp/issues/198#issuecomment-329305796

zzzpotato committed:
Formatting

zzzpotato committed:
Merge branch 'master' into Coin-config-additions-V2

Support for example directory

zzzpotato committed:
Moved pool_configs to examples directory

zzzpotato committed:
Deleted vot.json, xsg.json, zel.json from pool_configs

Gumbees commented on pool_configs/examples/vot.json:
```diff
@@ -0,0 +1,90 @@
+            "host": "127.0.0.1",
+            "port": 8232,
+            "user": "rpcuser",
+            "password": "prcpass"
```
Typo.. should be rpcpassword

Gumbees commented on pool_configs/examples/xsg.json:
```diff
@@ -0,0 +1,86 @@
+            "host": "127.0.0.1",
+            "port": 16112,
+            "user": "rpcuser",
+            "password": "rpcpass"
```
rpcpassword

Gumbees commented on pool_configs/examples/xsg.json:
```diff
@@ -0,0 +1,86 @@
+            "host": "127.0.0.1",
+            "port": 16112,
+            "user": "rpcuser",
+            "password": "rpcpass"
```
rpcpassword

Gumbees commented on pool_configs/examples/xsg.json:
```diff
@@ -0,0 +1,86 @@
+        "port": 3306,
+        "user": "me",
+        "password": "mypass",
+        "database": "zec",
```
Going to ignore this for now since it will be removed. For all pool_configs.

Gumbees committed:
Update vot.json

Gumbees committed:
Update zel.json

Gumbees committed:
Update xsg.json

Gumbees committed:
Update vot.json

Gumbees committed:
Update xsg.json

Gumbees committed:
Update zec.json

Gumbees committed:
Update zel.json

Gumbees committed:
Update zen.testnet.json

Gumbees committed:
Update zec.testnet.json

Gumbees committed:
Update xsg.json

Gumbees reviewed (dismissed):
Reviewed all pool_config/examples and standardized the syntax. Mainly rpcuser and rpcpassword and the MPOS database name.

egyptianbman committed:
A few more adjustments
